### ISSUE TITLE
Modules for scrnaseq

### ISSUE THREAD

**Issue by ggabernet** [nf-core/scrnaseq#50]
Needed modules for this pipeline:
- [x] gffread
- [x] salmon index
- [x] STAR genomegenerate
- [x] STAR align
- [x] Samtools index
- [x] kallisto index
- [x] kallisto quant
- [ ] build gene map kallisto (custom?)
- [x] build txp2gene (using bioawk, awk, custom?)
- [x] salmon alevin, see this https://github.com/nf-core/modules/pull/397
- [x] alevin_qc
- [x] kallisto bus
- [ ] https://github.com/nf-core/modules/issues/946
- [ ] bustools count
- [x] kb modules (kb/ref, kb/count)

**Comment by KevinMenden** [comment]
Made a draft PR for `salmon alevin` here: https://github.com/nf-core/modules/pull/397
So I checked it off the list. Works locally with some test data but that's too big for uploading so I'm keeping that as draft for now until we know about what we'll use as scRNA-seq test data.

**Comment by FloWuenne** [comment]
PRs for [kallistobustools/ref](https://github.com/nf-core/modules/pull/408) and [kallistobustools/count](https://github.com/nf-core/modules/pull/409) are also up and passing all tests. Thanks, @KevinMenden for the quick reviews!

**Comment by apeltzer** [comment]
Buried in projects at the moment, sorry folks for not being as responsive as during the hackathon :-( 
Will try to catch up once I'm a bit freed up again

**Comment by KevinMenden** [comment]
I just checked what is still missing, I think there is an open PR for `kallisto/quant` here:
https://github.com/nf-core/modules/pull/351
So no need to do that again. Will have a look later and see what's missing there.

I think the `build_gene_map` might be best to implement as a local process, what do you think?
I have modified the `gffread` module to handle the txp2gene thing, so I checked that off as well.

STAR and samtools should be possible via available modules, right? 🤔 
Then it would be just `alevin_qc` and the kallisto/bustools modules left. I can have a go at `alevin_qc` now. Is `kallisto bus` and `bustools` not already  implemented by the `kb` module in a way?

**Comment by ggabernet** [comment]
yes, we will not need `kallisto bus` and `bustools` for now as the `kb` modules are there

**Comment by BatoolMM** [comment]
Sorry for the delay with this PR nf-core/modules#351. I've been swamped with work in the lat couple of weeks. I'll get it done and ready for review in the weekend, is that alright? If you need it sooner, I can work on it tonight.

**Comment by KevinMenden** [comment]
Hi @BatoolMM, sometime next week would be awesome, but not need to stress yourself, there are enough other modules/things to finish :+1:

**Comment by grst** [comment]
Would be nice to turn all remaining local modules into nf-core modules before the next release!

**Comment by apeltzer** [comment]
Doesn't look like we need these anymore: Kallisto + Bustools are now handled by that python wrapper app (kbtools), which means we won't have to add additional modules to nf-core/modules apart from that (which is already there).

**Comment by grst** [comment]
There's still a bunch of local modules, though. Would be nice if as many of those as possible could be replaced with modules from the modules repo.

**Comment by apeltzer** [comment]
Yeah fair enough
Could move these upstream 👍🏻
